app/src/main/java/ca/pkay/rcloneexplorer/Rclone.java

```diff
@@ -59,55 +60,51 @@ public Rclone(Context context) {
         log2File = new Log2File(context);
     }
 
+    private String[] createCommand(ArrayList<String> args) {
+        String[] command = new String[args.size()];
+        for (int i = 0; i < args.size(); i++) {
+            command[i]= args.get(i);
+        }
+        return command;
+    }
+
     private String[] createCommand(String ...args) {
         boolean loggingEnabled = PreferenceManager
                 .getDefaultSharedPreferences(context)
                 .getBoolean(context.getString(R.string.pref_key_logs), false);
-        int staticArgSize = loggingEnabled ? 4 : 3;
-        int arraySize = args.length + staticArgSize;
-        String[] command = new String[arraySize];
+        ArrayList<String> command = new ArrayList<>();
 
-        command[0] = rclone;
-        command[1] = "--config";
-        command[2] = rcloneConf;
+        command.add(rclone);
+        command.add("--config");
+        command.add(rcloneConf);
 
         if(loggingEnabled) {
-            command[3] = "-vvv";
-        }
-
-        int i = staticArgSize;
-        for (String arg : args) {
-            command[i++] = arg;
+            command.add("-vvv");
         }
-        return command;
+        command.addAll(Arrays.asList(args));
+        return createCommand(command);
     }
 
     private String[] createCommandWithOptions(String ...args) {
         boolean loggingEnabled = PreferenceManager
                 .getDefaultSharedPreferences(context)
                 .getBoolean(context.getString(R.string.pref_key_logs), false);
-        int staticArgSize = loggingEnabled ? 8 : 7;
-        int arraySize = args.length + staticArgSize;
-        String[] command = new String[arraySize];
+        ArrayList<String> command = new ArrayList<>();
         String cachePath = context.getCacheDir().getAbsolutePath();
 
-        command[0] = rclone;
-        command[1] = "--cache-chunk-path";
-        command[2] = cachePath;
-        command[3] = "--cache-db-path";
-        command[4] = cachePath;
-        command[5] = "--config";
-        command[6] = rcloneConf;
+        command.add(rclone);
+        command.add("--cache-chunk-path");
+        command.add(cachePath);
+        command.add("--cache-db-path");
+        command.add(cachePath);
+        command.add("--config");
+        command.add(rcloneConf);
 
         if(loggingEnabled) {
-            command[7] = "-vvv";
+            command.add("-vvv");
         }
-
-        int i = staticArgSize;
-        for (String arg : args) {
-            command[i++] = arg;
-        }
-        return command;
+        Collections.addAll(command, args);
+        return createCommand(command);
```

Comment:
toArray(new String[0]); ?

Reply:
Initially i edited all the commands so that rclone would take longer to finish ('--bwlimit')
Switching away from arrays supposedly made this easier, but in the end it didn't really matter. However, i think this way is a little less error-prone when a dev edits code here, since he or she does not need to take the arraysize into account. It reduces complexity a tiny bit. So i kept it. Or we can undo it if you want, im fine with it either way

Reply:
Okay, then lets keep the local ArrayList.

Can you replace the last line with `return command.toArray(new String[0]);` as requested?